### DISCUSSION TITLE
command-instead-of-module: systemctl used in place of systemd module

### DISCUSSION THREAD

**Discussion by rclilly** [ansible/ansible-lint#4086] [Q&A]
Three years ago this pull request was merged: https://github.com/ansible/ansible-lint/pull/1544. It was supposed to allow the use of `'systemctl': ['set-default', 'show-environment']`, etc. since the [systemd](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_service_module.html) module does not have that functionality. However, when I run **ansible-lint** against my playbook it's still flagging that with the message: **_command-instead-of-module: systemctl used in place of systemd module._** Has that change been removed and the functionality added to the [systemd](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_service_module.html) module? What is the recommended way to determine the default runlevel and to set it to the desired, e.g. `systemctl set-default multi-user`?

**Comment by konstruktoid** [comment]
Can't reproduce using `systemctl set-default graphical.target`. `systemctl get-default` is however missing.

```sh
~$ ansible-lint --version
ansible-lint 24.2.1 using ansible-core:2.16.5 ansible-compat:4.1.11 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
~$ cat test.yml 
---
- name: Testing
  hosts: localhost
  any_errors_fatal: true
  gather_facts: false
  tasks:
    - name: Set multiuser
      ansible.builtin.command:
        cmd: systemctl set-default graphical.target
      register: set_default
      changed_when: set_default.rc == 0
~$ ansible-lint test.yml 

Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
```

**Reply by rclilly** [reply]
You are correct, the failure doesn't appear with `systemctl set-default multiuser.target`. It **_is_** showing for `systemctl get-default`.

**Reply by konstruktoid** [reply]
https://github.com/ansible/ansible-lint/pull/4087
